roles/cockpit-ui/tasks/main.yml

```diff
@@ -22,54 +22,64 @@
   changed_when: false
 
 - name: Create passthrough route for docker-registry
-  command: >
-    {{ openshift.common.client_binary }} create route passthrough
-    --service docker-registry
-    --config={{ openshift_hosted_kubeconfig }}
-    -n default
-  register: create_docker_registry_route
-  changed_when: "'already exists' not in create_docker_registry_route.stderr"
-  failed_when: "'already exists' not in create_docker_registry_route.stderr and create_docker_registry_route.rc != 0"
-  when: check_docker_registry_exists.rc == 0
+  oc_route:
+    kubeconfig: "{{ openshift_hosted_kubeconfig }}"
+    name: docker-registry
+    namespace: default
+    service_name: docker-registry
+    state: present
+    tls_termination: passthrough
+  run_once: true
 
 - name: Create passthrough route for registry-console
-  command: >
-    {{ openshift.common.client_binary }} create route passthrough
-    --service registry-console
-    --port registry-console
-    --config={{ openshift_hosted_kubeconfig }}
-    -n default
-  register: create_registry_console_route
-  changed_when: "'already exists' not in create_registry_console_route.stderr"
-  failed_when: "'already exists' not in create_registry_console_route.stderr and create_registry_console_route.rc != 0"
-  when: check_docker_registry_exists.rc == 0
+  oc_route:
+    kubeconfig: "{{ openshift_hosted_kubeconfig }}"
+    name: registry-console
+    namespace: default
+    service_name: registry-console
+    state: present
+    tls_termination: passthrough
+  run_once: true
 
-- name: Retrieve docker-registry route
-  command: >
-    {{ openshift.common.client_binary }} get route docker-registry
-    -o jsonpath='{.spec.host}'
-    --config={{ openshift_hosted_kubeconfig }}
-    -n default
-  register: docker_registry_route
+- name: Retrieve docker-registry json
```

Comment:
@detiber / @mtnbikenc this seems really clunky to me. I'm positive there is a simpler way to accomplish this but I just have not needed to do it before. Any pointers?